src/messages.cpp

```diff
@@ -77,8 +104,13 @@ bytes
 GroupInfo::to_be_signed() const
 {
   tls::ostream w;
-  w << group_id << epoch << tree << confirmed_transcript_hash
-    << interim_transcript_hash << path << confirmation << signer_index;
+  tls::vector<1>{}.encode(w, group_id);
+  w << epoch << tree;
+  tls::vector<1>{}.encode(w, confirmed_transcript_hash);
+  tls::vector<1>{}.encode(w, interim_transcript_hash);
+  w << path;
+  tls::vector<1>{}.encode(w, confirmation);
+  w << signer_index;
   return w.bytes();
```

Comment:
ah ! this side effect :-)

Reply:
Yeah, it's ugly.  The trick here is that you need something ternary to combine (a) the stream (b) the trait and (c) the value.  You might be able to do this with `iomanip`-style [stream updaters](http://www.cplusplus.com/reference/iomanip/),  But then you need to (temporarily) persist the fact that there's a trait waiting to be processed, which could be non-trivial.  Or you could make temporary wrapper objects, but that's  challenging to do without causing additional allocations (requires the wrapper object to have a pointer or reference to the underlying data).  

```
// iomanip
w << path 
  << tls::vector<1>() << interim_transcript_hash
  << signer_index;

// wrapper
w << path << tls::vector<1>(confirmation) << signer_index;
```

After pondering a few of these options, my feeling was that having the traits as they are in the PR struck the right balance between simplicity and clarity in the default case (`TLS_TRAITS`).